src/Fantomas/CodePrinter.fs

```diff
@@ -681,48 +681,129 @@ and genMemberBinding astContext b =
             genMemberFlagsForMemberBinding astContext mf b.RangeOfBindingAndRhs
             +> ifElse isInline (!- "inline ") sepNone +> opt sepSpace ao genAccess
 
+        let rec genNameAndParameters p =
+            match p with
+            | PatLongIdent(ao, s, ps, tpso) ->
+                let aoc = opt sepSpace ao genAccess
+                let tpsoc = opt sepNone tpso (fun (ValTyparDecls(tds, _, tcs)) -> genTypeParamPostfix astContext tds tcs)
+                let s = if s = "``new``" then "new" else s
```

Comment:
I think I don't understand why this would be necessary - could you fill me in please?

Reply:
Normally the pattern expression of the member would be composed in [genPat](https://github.com/fsprojects/fantomas/blob/515ce2ee6beaab631df9a47e0c9fd945c04e7283/src/Fantomas/CodePrinter.fs#L2971). There the name of the member and parameters would be printed.

To adhere to both style guides I needed more control over this. 
I could not control certain newline decisions in `genPat` as there are specific to the member binding.
`genPat` is used all over the place so that is where that extra `new` check will somewhere be useful.

As I copied over some of the code from `genPat` to `genNameAndParameters` I didn't really think about this.
I can indeed remove the extra `let s = if ...` line.